flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -365,11 +368,9 @@ class ExprLowering {
     }
     auto name = converter.mangleName(*symbol);
     auto func = builder.getNamedFunction(name);
-    // TODO: If this is an external not called/defined in this file
-    // (e.g, it is just being passed as a dummy procedure argument)
-    // we need to create a funcOp for it with the interface we have.
     if (!func)
-      TODO();
+      func = Fortran::lower::declareFunction(proc, converter,
+                                             llvm::StringRef{name});
```

Comment:
StringRef has a number of implicit conversions, so this ctor can probably be omitted.

Reply:
The argument was actually an `llvm::Optional<llvm::StringRef>`, so C++ needs to be guided a bit. However, I removed the Optional aspect, so this explicit ctor is removed.